components/eamxx/scripts/machines_specs.py

```diff
@@ -199,10 +199,20 @@ def get_mach_testing_resources(machine):
     of jobs across cores.
     """
     if is_cuda_machine(machine):
-        return int(run_cmd_no_fail("nvidia-smi -L | wc -l"))
+        prefix = "srun " if is_salloc(machine) else ""
+        return int(run_cmd_no_fail(f"{prefix}nvidia-smi -L | wc -l"))
```

Comment:
This was subtle. Salloc + python does not necessarily run commands on the compute node unless `srun` is provided.